src/control/server/mgmt_pool.go

```diff
@@ -543,20 +543,24 @@ func (svc *mgmtSvc) PoolDestroy(ctx context.Context, req *mgmtpb.PoolDestroyReq)
 		ds := drpc.DaosStatus(evresp.Status)
 		svc.log.Debugf("MgmtSvc.PoolDestroy drpc.MethodPoolEvict, evresp:%+v\n", evresp)
 
+		// If the destroy request is being forced, we should additionally zap the label
+		// so the entry doesn't prevent a new pool with the same label from being created.
+		if req.Force {
+			ps.PoolLabel = ""
```

Comment:
Just for my understanding, what will happen to the meta-data information associated to the label already existing ?
All the data will be overwritten => no space leask needing a kind of Garbage Collection ?

Reply:
Go is a garbage collected language, so it's almost never necessary to do any explicit memory management. Setting the `ps.PoolLabel` variable to the empty string value will result in the existing string being marked as unreachable and the GC will free its memory on the next GC pass.

The `svc.sysdb.UpdatePoolService()` call is responsible for creating a raft update which is applied to the service backing the MS DB so that the change to the pool metadata is distributed to all replicas.

Reply:
In fact, I was not asking at the go level but at the level of the metadata stored in the SCM. It is probably a dumb question, but from my understanding the information of the pool associated to this label are stored in the SCM. Thus, I was thinking that with this patch, we are going to delete this association and thus have some dangling metadata stored in the SCM.

Reply:
Sorry, still drinking my coffee. ;) I think maybe you are asking about the pool service entry left behind? If so, then what we are doing is making the current pool service entry what I call a "tombstone" entry, i.e. it's dead. These entries will eventually be cleaned up by [checkPools()](https://github.com/daos-stack/daos/blob/master/src/control/server/mgmt_pool.go#L453) which gets called on Leader step-up.

Reply:
OK, so from what I understand the `checkPools()` function will be eventually called and will do the garbage collection of the "tombstone" entries such `git gc`. Thanks for the explanation.

Reply:
Yes. And please note, this tombstone/cleanup solution is only for the case where the destroy request has been forced and the `PoolDestroy()` handler returns early due to an engine error. It's a little messy, but we want to avoid a situation where the admin cannot destroy a pool due to some problem with the pool service. I think there is more work to be done to handle this situation.